standards.md

```diff
@@ -1,47 +1,126 @@
-# Tekton Development Standards
+# Tekton Pipelines Contributor and Reviewer Expectations
 
-This doc contains the standards we try to uphold across all project
-in the Tekton org.
+The purpose of this doc is to:
 
-## Principles
+* Outline for contributors what we expect to see in a pull request
+* Establish a baseline for reviewers so they know at a minimum what to look for
 
-When possbile, try to practice:
+Each Pull Request is expected to meet the following expectations around:
 
-- **Documentation driven development** - Before implementing anything, write
-  docs to explain how it will work
-- **Test driven development** - Before implementing anything, write tests to
-  cover it
+* [Pull Request Description](#pull-request-description)
+* [Commits](#commits)
+* [Docs](#docs)
+* [Tests](#tests)
+* [API Design](#api-design)
+* [Functionality](#functionality)
+* [Content](#content)
+* [Code](#code)
 
-Minimize the number of integration tests written and maximize the unit tests!
-Unit test coverage should increase or stay the same with every PR.
+_See also [the Tekton review process](https://github.com/tektoncd/community/blob/master/process.md#reviews)._
 
-This means that most PRs should include both:
+## Pull request description
 
-1. Tests
-2. Documentation updates
+* Include a link to the issue being addressed, but describe the context for the reviewer
+  * If there is no issue, consider whether there should be one:
+    * New functionality must be designed and approved, may require a TEP
+    * Bugs should be reported in detail
+  * Release notes filled in for user visible changes (bugs + features),
+    or removed if not applicable (refactoring, updating tests)
```

Comment:
NIT: I believe with the release notes plugin now for no release notes we need to set "NONE" into the release notes block, but it may be ok to not go into this level of detail here.

Reply:
good point !

Reply:
(although, the plugin is "only" enabled on `pipeline`, `cli` and `operator` so far)

Reply:
I've added a note about this also! Over time as we automate more maybe it will make sense to clearly identify which parts are automated and link to supporting docs + config